### PR TITLE
Add cards for authorizing and connecting to GitHub

### LINKED ISSUE #70376: Create the Authorize UI and GitHub integration panel
### Details

In addition to the GitHub connection option at `/marketing/connections/:site` the user should be able to connect GitHub via the `hosting-config` page. The URL used on the hosting page `Authorize` button should be the same as the `connect` button on the marketing page. 

connect_github

### Checklist

- [ ] UI looks like mockups
- [ ] Works for RTL
- [ ] Can connect GitHub via the hosting page and the connection shows in the `/me/connections` endpoint response

### Related

_No response_

### PR DESCRIPTION
**Pull request by mpkelly**
**Update**: the changes included in this PR now go beyond the scope of the original, so I have made it a draft. I will likely abandon it in favour of some smaller PRs later. 

---

Closes https://github.com/Automattic/wp-calypso/issues/70376

See also D98139-code

#### Proposed Changes

* Add a new card to the hosting page that can be used to start the GitHub auth flow. 

image

 Notes:
-  this card is always shown for now, but it should be shown conditionally according to if we are already connected or not
- The copy isn't final. Proposals welcome
- It implements the mockup UI, but we will want to link to docs and stuff in the final version (docs update is covered by another task)
- It uses the WordPress app, which is owned by my account currently but will be transferred to Automattic later https://github.com/apps/wordpress-dot-com

#### Testing Instructions

* Apply patch D98139-code
* Make sure the API is sandboxed
* Checkout this branch local
* Go to "Hosting configuration" for an Atomic site
* Click authorise and you should be redirected to somewhere like https://public-api.wordpress.com/rest/v1.2/github-authorize-callback?siteId=210341954&code=0922b1371c663ecc5092, which is expected until more endpoints are added. 
* Verify the `siteId` and `code` are present in the URL

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #

### CODE REVIEW

File: client/my-sites/hosting/github-authorize-card/index.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import { getSelectedSiteId } from 'calypso/state/ui/selectors/index';
+import iconGitHub from './github.svg';
+
+import './style.scss';
+
+export const GithubAuthorizeCard = () => {
+	const siteId = useSelector( getSelectedSiteId );
+	return (
+		<Card className="github-authorize-card">
+			<img className="github-icon" src={ iconGitHub } alt="" />
```
Comment (p-jackson) [inline client/my-sites/hosting/github-authorize-card/index.tsx]: I don't think `github-icon` is namespaced enough, I can imagine something else in Calypso using this name too. I was thinking something like `github-authorize-icon`.

File: client/my-sites/hosting/github-authorize-card/index.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+			<CardHeading>{ translate( 'Connect GitHub' ) }</CardHeading>
+			<p>{ translate( 'Connect to a GitHub repository and branch and deploy directly.' ) }</p>
```
Comment (p-jackson) [inline client/my-sites/hosting/github-authorize-card/index.tsx]: What do you think about:
`Connect this site to a GitHub repository, choose a branch, and deploy with each push.`

**Review by p-jackson — COMMENTED** [review]
> Verify the `siteId` and `code` are present in the URL

I don't see `code`, but I wouldn't expect it either. Wouldn't the code be kept on the server side?

How are you planning to prevent users from seeing this UI till we ship? Are we going to have a feature flag?

**Comment by mpkelly** [comment]
> I don't see `code`, but I wouldn't expect it either. Wouldn't the code be kept on the server side?

You should be directed to an endpoint that doesn't exist, and this URL should include the `code`, e.g. 

https://public-api.wordpress.com/rest/v1.2/github-authorize-callback?siteId=210341954&code=cab0ddeefcf1965067f1

Once added, that endpoint will consume the `code` and then redirect to the frontend with the `code` removed. Is the API sandboxed? I need to add that to the description. 

I forgot about the feature flag. I will add that now!

File: client/my-sites/hosting/github-authorize-card/index.tsx (hunk 1 above)
Comment (mpkelly) [reply client/my-sites/hosting/github-authorize-card/index.tsx]: Ok, will do that.

File: client/my-sites/hosting/github-authorize-card/index.tsx (hunk 2 above)
Comment (mpkelly) [reply client/my-sites/hosting/github-authorize-card/index.tsx]: Works for me.

**Comment by mpkelly** [comment]
Just so it's clear, there is also task https://github.com/Automattic/wp-calypso/issues/70375, which will handle the GitHub request from the authorization flow and exchange the `code` for an `access_token`. That task is more server focused while this is UI-focused.

**Comment by p-jackson** [comment]
Superseded by https://github.com/Automattic/wp-calypso/pull/72625

**Pull request closed without merging**